Mathlib/GroupTheory/Exponent.lean

```diff
@@ -349,8 +357,9 @@ section CommMonoid
 variable [CommMonoid G]
 
 @[to_additive]
-theorem exists_orderOf_eq_exponent (h : ∀ g : G, 0 < orderOf g) (he : exponent G ≠ 0) :
+theorem exists_orderOf_eq_exponent (h : ∀ g : G, 0 < orderOf g) (hG : ExponentExists G) :
```

Comment:
This first condition is redundant
```suggestion
theorem exists_orderOf_eq_exponent (hG : ExponentExists G) :
```

Reply:
Only if you assume cancellativity, as far as I can tell. If not, please prove :)

Reply:
This is true in *any* monoid, almost by definition. I would put these right before `Monoid.exponent_ne_zero`.
```lean
@[to_additive]
theorem ExponentExists.isOfFinOrder (h : ExponentExists G) (g : G) : IsOfFinOrder g :=
  isOfFinOrder_iff_pow_eq_one.mpr <| by peel 2 h; exact this g

@[to_additive]
theorem ExponentExists.orderOf_pos (h : ExponentExists G) (g : G) : 0 < orderOf g :=
  h.isOfFinOrder g |>.orderOf_pos
```
you can use the latter to golf this proof too:
```lean
theorem exponent_eq_zero_of_order_zero {g : G} (hg : orderOf g = 0) : exponent G = 0 :=
  exponent_eq_zero_iff.mpr fun h ↦ h.orderOf_pos g |>.ne' hg
```